### PR TITLE
validate: add namespace if is duplicated check

### PR DESCRIPTION
**Pull request by Mashimiao**
Signed-off-by: Ma Shimiao <mashimiao.fnst@cn.fujitsu.com>

### CODE REVIEW

File: validate/validate.go [hunk 1]
```diff
@@ -309,31 +309,40 @@ func (v *Validator) CheckMounts() (msgs []string) {
...
-				netExists = true
-			} else if v.spec.Linux.Namespaces[index].Type == rspec.MountNamespace {
-				mountExists = true
-			} else if v.spec.Linux.Namespaces[index].Type == rspec.UserNamespace {
-				userExists = true
-			}
 		}
+
+		nsType := v.spec.Linux.Namespaces[index].Type
+		nsPath := v.spec.Linux.Namespaces[index].Path
```
Comment (wking) [inline validate/validate.go]: nit: I'd rather have `ns := v.spec.Linux.Namespaces[index]`.  Only one variable (vs. the two here), and you only need an extra character when you use it (e.g. `ns.Type` vs your `nsType`).

**Comment by wking** [comment]
de21837a looks good to me, although I'd slightly prefer [a single convenience variable][1].

Do we want to link [here][2] from the error message?  Or are we punting on that until we're ready for some future push on #16?

[1]: https://github.com/opencontainers/runtime-tools/pull/295#discussion_r93669755
[2]: https://github.com/opencontainers/runtime-spec/blob/v1.0.0-rc3/config-linux.md#namespaces

File: validate/validate.go (hunk 1 above)
Comment (Mashimiao) [inline validate/validate.go]: fixed.

**Comment by Mashimiao** [comment]
@wking  I think we'd better don't link from error message in this PR. If we want to add links, we should find out all error message related with spec. part of messages have link, part of not, that does not look unified.  And maybe adding links will cause maintaining problem, that needs more discussion

**Comment by Mashimiao** [comment]
@hqhq  @mrunalp  @liangchenye  PTAL

**Comment by Mashimiao** [comment]
ping @opencontainers/runtime-tools-maintainers

**Comment by liangchenye** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**